### ISSUE TITLE
Symbols inside quoted captions affect parsing of options

### ISSUE THREAD

**Issue by ralmond** [quarto-dev/quarto#211]
Putting punctuation (e.g., `=` or `,`) inside of a caption screws up parsing, even though it is inside quotes.

````
```{r}
#| label: tbl-tab
#| tbl-cap:  "A 3-way array"
#| tbl-subcap:
#|   - "Var3 = 1"
#|   - "Var3 = 2"
arr <- array(1:24,c(4,3,2))
arr[,,1]
arr[,,2]
```
Look at Table @tbl-tab, just look at it.
````

This code issues the following warnings:
```
Warning messages:
1: In value[[3L]](cond) : Failed to parse chunk options in body:
Error in parse(text = code, keep.source = FALSE): <text>:1:24: unexpected symbol
1: alist( label: tbl-tab1 tbl
                           ^

2: In opts$engine <- unlist(strsplit(rmdChunkOpts, split = "(\\s|,)+"))[[1]] :
  Coercing LHS to a list
```
It then complains about not being able to find cross-reference `@tbl-tab`.
Removing the equal signs fixes the problem (at the expense of legibility of the captions).  Note that inserting a comma in a caption generates a similar issue.

I'm running Quarto version 1.0.37 under R studio:
"Mountain Hydrangea" Release (583b465e, 2023-06-06) for Ubuntu Jammy
Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) rstudio/2023.06.0+421 Chrome/110.0.5481.208 Electron/23.3.0 Safari/537.36

**Comment by jjallaire** [comment]
@cderv I don't see this issue w/ the latest Quarto (and noted that the version reported here is fairly old, 1.0.37). I think this is related to Knitr attempting to parse for single-line options. Is this an issue w/ Knitr or something we've subsequently fixed in Quarto.

@ralmond What version of Knitr are you using?

**Comment by cderv** [comment]
I think knowing the **knitr** version will definitely help. I don't see the error either, and I don't remember a specific fix for that in Quarto. So possibly fix in **knitr** directly... 

@ralmond is updating Quarto possible ? 


FWIW 
> 2: In opts$engine <- unlist(strsplit(rmdChunkOpts, split = "(\\s|,)+"))[[1]] :
  Coercing LHS to a list

This is coming for RStudio IDE 
https://github.com/rstudio/rstudio/blob/a9a0652e3feac02a05caeead5d54e370f3ec4209/src/cpp/session/modules/SessionRmdNotebook.R#L686

**Comment by ralmond** [comment]
My knitr version is 1.43.  `update.packages()` is not showing a newer version.

Is there an update function on Quarto?  I'm showing this as a manual install, so I wasn't aware there was a new version.  (Nice if it could get in the Debian stream, so it would be picked up by Ubuntu & Pop OS).

**Comment by cderv** [comment]
> In particular, the issue is happening when I run the chunk in RStudio, so I guess Quarto isn't even being invoked.

Yes I can reproduce the warning while running interactively inside RStudio IDE. So the issue would be best open in [rstudio/rstudio](https://github.com/rstudio/rstudio) repo 

> It then complains about not being able to find cross-reference @tbl-tab.

@ralmond regarding this, you need to output a table for the cross reference on table to work. Array is not output as table by **knitr** AFAIK

Try to render by converting your array to a Table output. 
````markdown
```{r}
#| label: tbl-tab
#| echo: false
#| tbl-cap: "A 3-way array"
#| tbl-subcap:
#|   - "Var3 1"
#|   - "Var3 = 2"
arr <- array(1:24,c(4,3,2))
knitr::kable(arr[,,1])
knitr::kable(arr[,,2])
```
````

Crossreferencing using `@tbl-` and other `tbl-` feature as option requires a Table. Not R code output.
